pkg/registry/apps/statefulset/strategy_test.go

```diff
@@ -204,3 +209,81 @@ func TestStatefulSetStatusStrategy(t *testing.T) {
 		t.Errorf("unexpected error %v", errs)
 	}
 }
+
+func makeStatefulSetWithMaxUnavailable(maxUnavailable *int) *apps.StatefulSet {
+	rollingUpdate := apps.RollingUpdateStatefulSetStrategy{}
+	if maxUnavailable != nil {
+		maxUnavailableIntStr := intstr.FromInt(*maxUnavailable)
+		rollingUpdate = apps.RollingUpdateStatefulSetStrategy{
+			MaxUnavailable: &maxUnavailableIntStr,
+		}
+	}
+
+	return &apps.StatefulSet{
+		Spec: apps.StatefulSetSpec{
+			UpdateStrategy: apps.StatefulSetUpdateStrategy{
+				Type:          apps.RollingUpdateStatefulSetStrategyType,
+				RollingUpdate: &rollingUpdate,
+			},
+		},
+	}
+}
+
+func TestDropDisabledField(t *testing.T) {
```

Comment:
Since we have time, make these tests part of `TestStatefulSetStrategy` instead of testing a non-exported function.

Reply:
I agree with @alculquicondor. I think it'd be better if we have test functions for both `PrepareForCreate` and `PrepareForUpdate`

Reply:
i am not sure i agree with this @alculquicondor @ravisantoshgudimetla  . I think if there is a common functionality which is the least common denominator across many exported functions, it might be easier to test that common functionality irrespective of whether its exported or not. If we think there are more tests we can add at the PrepareForCreate/Update level also , let me know what you folks think

Reply:
The question that this test doesn't answer: What guarantees that the exported function is calling the helper method?

It's a common practice in golang to test exported functions unless it's really hard to do so.

OTOH, I agree that there should be separate tests for `PrepareForCreate` and `PrepareForUpdate`. I'm not familiar with the existing tests, so I'm not sure if it's a huge amount of work to separate them. If it is, leave it for a follow up.

Reply:
Fair enough, if you think it is too much of work, please add a TODO and leave it. If it helps, I added units for creation/updation for the registry changes [here](https://github.com/kubernetes/kubernetes/pull/100842/commits/f3ab9550180a57e693105f77e8eba1fa00499fe2#diff-bfde8b1db7251cfd89bbce04b4397ea0b7579b7988ff4adbbb1725cbf06312c3L29)


We already have [test](TestDropDisabledField) to test the same functionality - can we combine both?

Reply:
I have added a TODO and will look at this after I have converted the main tests to table driven

Reply:
The above comment of combining `TestDropDisabledField` and `TestDropStatefulSetDisabledFields` is still unanswered

Reply:
This still applies as well.